pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -1599,15 +1599,22 @@ public ResultSet getSchemas() throws SQLException {
   public ResultSet getSchemas(@Nullable String catalog, @Nullable String schemaPattern)
       throws SQLException {
     String sql;
-    sql = "SELECT nspname AS \"TABLE_SCHEM\", NULL AS \"TABLE_CATALOG\" FROM pg_catalog.pg_namespace "
-          + " WHERE nspname <> 'pg_toast' AND (nspname !~ '^pg_temp_' "
-          + " OR nspname = (pg_catalog.current_schemas(true))[1]) AND (nspname !~ '^pg_toast_temp_' "
-          + " OR nspname = replace((pg_catalog.current_schemas(true))[1], 'pg_temp_', 'pg_toast_temp_')) ";
+    sql = "SELECT schema_name AS \"TABLE_SCHEM\", catalog_name AS \"TABLE_CATALOG\" FROM information_schema.schemata "
+          + " WHERE schema_name <> 'pg_toast' AND (schema_name !~ '^pg_temp_' "
+          + " OR schema_name = (pg_catalog.current_schemas(true))[1]) AND (schema_name !~ '^pg_toast_temp_' "
+          + " OR schema_name = replace((pg_catalog.current_schemas(true))[1], 'pg_temp_', 'pg_toast_temp_')) ";
+    if (catalog != null) {
+      if (catalog.isEmpty()) {
+        sql += " AND false ";
```

Comment:
Why not ignore this case like the NULL case?

Reply:
If `catalog` is `""`, then getSchemas should return schemas "without a catalog" per [JDBC spec:](https://docs.oracle.com/en/java/javase/17/docs/api/java.sql/java/sql/DatabaseMetaData.html#getSchemas(java.lang.String,java.lang.String))

> catalog - a catalog name; must match the catalog name as it is stored in the database;"" retrieves those without a catalog; null means catalog name should not be used to narrow down the search.